src/org/zaproxy/zap/extension/search/ExtensionSearch.java

```diff
@@ -71,9 +71,9 @@ public ExtensionSearch() {
 	@Override
 	public void hook(ExtensionHook extensionHook) {
 	    super.hook(extensionHook);
-	    extensionHook.addSessionListener(this);
 	    extensionHook.addOptionsParamSet(getSearchParam());
 	    if (getView() != null) {
+	        extensionHook.addSessionListener(this);

```

Comment:
I'm wondering if this could be a little confusing - you have to look at the sessionChanged method to check that its only relevant to the UI and not the daemon. What if someone changes this to do something else? OK, we should then check that its initialized in daemon mode but thats easily missed. 
2 suggestions - either add suitable comments (above and in sessionChanged?) or change sessionChangedEventHandler to only call this.getSearchPanel().resetSearchResults(); if View.isInitialized()